sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -46,9 +46,13 @@ class QueryExecution(val sparkSession: SparkSession, val logical: LogicalPlan) {
   protected def planner = sparkSession.sessionState.planner
 
   def assertAnalyzed(): Unit = {
-    try sparkSession.sessionState.analyzer.checkAnalysis(analyzed) catch {
+    var analyzedPlan: Option[LogicalPlan] = None
```

Comment:
@hvanhovell I tried the suggested code snippet. Actually since AnalyzedPlan is declared lazy the evaluation happens inside the try block and hence it has the same problem. So lets just move analyzed outside the try block as wenchen suggests ?

```scala
def assertAnalyzed(): Unit = {
    analyzed
    try {
      sparkSession.sessionState.analyzer.checkAnalysis(analyzed)
    } catch {
      case e: AnalysisException =>
        val ae = new AnalysisException(e.message, e.line, e.startPosition, Option(analyzed))
        ae.setStackTrace(e.getStackTrace)
        throw ae
    }
  }
```